### PR TITLE
docs: samples and tests for admin backup APIs

### PR DESCRIPTION
**Pull request by rahul2393**
Notes
* Adds samples and tests for auto-generated backup admin APIs.
* The sample code style is mostly consistent with our existing samples.

### CODE REVIEW

File: samples/samples/admin/backup_snippet.py [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # Start restoring an existing backup to a new database.
+    encryption_config = {
+        "encryption_type": RestoreDatabaseEncryptionConfig.EncryptionType.CUSTOMER_MANAGED_ENCRYPTION,
+        "kms_key_name": kms_key_name,
+    }
+    new_database = instance.database(
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:164]: Do we need this?

File: samples/samples/admin/backup_snippet.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.CreateBackupMetadata) "
+        "AND (metadata.database:{})"
+    ).format(database_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:258]: Should we instead directly use code used here? https://github.com/googleapis/python-spanner/blob/3aab0ed5ed3cd078835812dae183a333fe1d3a20/google/cloud/spanner_v1/instance.py#L726
Our handwritten code returns the data as `google.api_core.operation.Operation` and customers could use this for other things like getting the operationid, deserialize etc.

In this example the `metadata` is specific field of  `google.api_core.operation.Operation`. Instead directly showing the conversion to `google.api_core.operation.Operation` will help customers,
WDYT?

File: samples/samples/admin/backup_snippet.py [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/google.spanner.admin.database.v1.CopyBackupMetadata) "
+        "AND (metadata.source_backup:{})"
+    ).format(backup_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:277]: same here

File: samples/samples/admin/backup_snippet.py [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.OptimizeRestoredDatabaseMetadata)"
+    )
+    request = spanner_database_admin.ListDatabaseOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_database_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:307]: same here

File: samples/samples/admin/backup_snippet.py [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+
+    # Verify that the backup is deleted.
+    try:
+        backup = spanner_client.database_admin_api.get_backup(
+            backup_pb.GetBackupRequest(name=backup.name)
+        )
+    except NotFound:
+        print("Backup {} has been deleted.".format(backup.name))
+        return
+    assert True is False
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:438]: Why is this here?

File: samples/samples/admin/backup_snippet.py [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+    db = operation.result(1600)
+
+    # Newly created database has restore information.
+    restore_info = db.restore_info
+    print(
+        "Database {} restored to {} from backup {} with using encryption key {}.".format(
+            restore_info.backup_info.source_database,
+            new_database_id,
+            restore_info.backup_info.backup,
+            restore_info.encryption_config.kms_key_name,
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet.py:186]: Shoulkd this be `db. encryption_config`

File: samples/samples/admin/backup_snippet_test.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+@RetryErrors(exception=DeadlineExceeded, max_tries=2)
+def test_create_database_with_retention_period(capsys, sample_instance):
+    backup_snippet.create_database_with_version_retention_period(
+        sample_instance.instance_id,
+        RETENTION_DATABASE_ID,
+        RETENTION_PERIOD,
+    )
+    out, _ = capsys.readouterr()
+    assert (RETENTION_DATABASE_ID + " created with ") in out
+    assert ("retention period " + RETENTION_PERIOD) in out
```
Comment (harshachinta) [inline samples/samples/admin/backup_snippet_test.py:191]: make sure this database is dropped

File: samples/samples/admin/backup_snippet.py (hunk 2 above)
Comment (rahul2393) [reply samples/samples/admin/backup_snippet.py:258]: All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:
Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_backup_operations.go#L64

JAVA= https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1683-L1689

File: samples/samples/admin/backup_snippet.py (hunk 3 above)
Comment (rahul2393) [reply samples/samples/admin/backup_snippet.py:277]: All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:
Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_backup_operations.go#L64

JAVA= https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1683-L1689

File: samples/samples/admin/backup_snippet.py (hunk 4 above)
Comment (rahul2393) [reply samples/samples/admin/backup_snippet.py:307]: All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:

Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_database_operations.go#L58

Java=https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1735-L1742

File: samples/samples/admin/backup_snippet.py (hunk 5 above)
Comment (rahul2393) [reply samples/samples/admin/backup_snippet.py:438]: Success path should go into "except NotFound" exception we want to fail exception is not raised
Can you please suggest better way to do it?

File: samples/samples/admin/backup_snippet_test.py (hunk 7 above)
Comment (rahul2393) [reply samples/samples/admin/backup_snippet_test.py:191]: We are deleting the database in the sample itself.

File: samples/samples/admin/backup_snippet.py (hunk 5 above)
Comment (harshachinta) [reply samples/samples/admin/backup_snippet.py:435]: This looks wierd since it gets dispalyed on the public sample.
Can we instead assert `Backup {} has been deleted` this in the test? If this statement is there then all good, if not then it means delete backup failed.

**Review by harshachinta — APPROVED** [review]
